### PR TITLE
test/e2e: a bunch of small fixes and improvements.

### PR DESCRIPTION
**Pull request by klihub**
This patch set contains the following fixes to the end-to-end test infra:

  - fix a small typo in the usage/help message of `test/e2e/run_tests.sh`
  - correctly pass the `test`, `topology` and `policy` `directories` to `run.sh` for template resolution
  - print the resolved template path and any template shadowing `warnings` to `stderr`
  - catch and give special treatment to python `KeyError` and `IndexError` `exceptions` during verification
  - filter empty lines from evaluated YAML templates
  - don't hardcode a max. test case limit of 9 pods with 4 containers each

### CODE REVIEW

**Review by askervin — APPROVED** [review]
Great stuff! Started immediately developing on top of this!
LGTM

**Comment by klihub** [comment]
> Great stuff! Started immediately developing on top of this!
> LGTM

Thanks ! But argh..., now I spotted serious lysdexia in 4d82222a97f5f82d7dbacd9a31f29e1ccede4a11.
Please re-approve once the force-pushed fix lands.

**Review by askervin — APPROVED** [review]
LGTM

**Pull request merged**
